sdk/monitor/Azure.Monitor.OpenTelemetry.Exporter/tests/Azure.Monitor.OpenTelemetry.Exporter.Tests/ResourceProviderTests.cs

```diff
@@ -0,0 +1,91 @@
+{
+    public class ResourceProviderTests
+    {
+        [Fact]
+        public void Verify_GetResourceProviderDetails_Default()
+        {
+            var platform = new MockPlatform();
+            platform.OSPlatformName = "UnitTest";
+
+            var resourceProviderDetails = ResourceProvider.GetResourceProviderDetails(platform, new MockVmMetadataProvider());
+
+            Assert.Equal("unknown", resourceProviderDetails.ResourceProvider);
+            Assert.Equal("unknown", resourceProviderDetails.ResourceProviderId);
+            Assert.Equal("UnitTest", resourceProviderDetails.OperatingSystem);
+        }
+
+        [Fact]
+        public void Verify_GetResourceProviderDetails_AppService1()
+        {
+            var platform = new MockPlatform();
+            platform.OSPlatformName = "UnitTest";
+            platform.SetEnvironmentVariable("WEBSITE_SITE_NAME", "testWebSite");
+
+            var resourceProviderDetails = ResourceProvider.GetResourceProviderDetails(platform, new MockVmMetadataProvider());
+
+            Assert.Equal("appsvc", resourceProviderDetails.ResourceProvider);
+            Assert.Equal("testWebSite", resourceProviderDetails.ResourceProviderId);
+            Assert.Equal("UnitTest", resourceProviderDetails.OperatingSystem);
+        }
+
+        [Fact]
+        public void Verify_GetResourceProviderDetails_AppService2()
+        {
+            var platform = new MockPlatform();
+            platform.OSPlatformName = "UnitTest";
+            platform.SetEnvironmentVariable("WEBSITE_SITE_NAME", "testWebSite");
+            platform.SetEnvironmentVariable("WEBSITE_HOME_STAMPNAME", "testStampName");
+
+            var resourceProviderDetails = ResourceProvider.GetResourceProviderDetails(platform, new MockVmMetadataProvider());
+
+            Assert.Equal("appsvc", resourceProviderDetails.ResourceProvider);
+            Assert.Equal("testWebSite/testStampName", resourceProviderDetails.ResourceProviderId);
+            Assert.Equal("UnitTest", resourceProviderDetails.OperatingSystem);
+        }
+
+        [Fact]
+        public void Verify_GetResourceProviderDetails_Functions()
+        {
+            var platform = new MockPlatform();
+            platform.OSPlatformName = "UnitTest";
+            platform.SetEnvironmentVariable("FUNCTIONS_WORKER_RUNTIME", "test");
+            platform.SetEnvironmentVariable("WEBSITE_HOSTNAME", "testHostName");
+
+            var resourceProviderDetails = ResourceProvider.GetResourceProviderDetails(platform, new MockVmMetadataProvider());
+
+            Assert.Equal("functions", resourceProviderDetails.ResourceProvider);
+            Assert.Equal("testHostName", resourceProviderDetails.ResourceProviderId);
+            Assert.Equal("UnitTest", resourceProviderDetails.OperatingSystem);
+        }
+
+        [Fact]
+        public void Verify_GetResourceProviderDetails_AzureVM()
+        {
+            var platform = new MockPlatform();
```

Comment:
Not for this PR. But if possible we could do the [same](https://github.com/Azure/azure-sdk-for-net/pull/36816/files#r1218239402) for Platform as well.